server/models/listing.model.js

```diff
@@ -0,0 +1,50 @@
+const { PrismaClient } = require('@prisma/client')
+
+const prisma = new PrismaClient({log: ['query', 'info', 'warn', 'error'],  errorFormat: "minimal",}
+)
+
+async function newListing(id, petName){
+         //Check required properties
+    if (!id || !petName ){
+        return ({
+           error: 'Missing required properties.'
+        })
+    }
+    //Check if username doesn't exist already
+   //This check is done by enforcing a unique constraint on username(email) in the DB, done at the time of user creation
+
```

Comment:
@cris-valente Before attempting to create a listing via Prisma we have to perform a check on the userId provided.
If the provided userId has a role=shelter then they are allowed to create a listing
Otherwise we return an error : User does not have access rights for this.

So i would add another await method to find user by id (probably something like findUnique) and the check the returned record that indeed it has the shelter role
If the returned role is correct the we proceed to the create

Reply:
Yes, we were gonna address this issue now. Thanks Andra!